sources/engine/Stride.Graphics/Direct3D/QueryPool.Direct3D.cs

```diff
@@ -1,57 +1,97 @@
-﻿// Copyright (c) .NET Foundation and Contributors (https://dotnetfoundation.org/ & https://stride3d.net) and Silicon Studio Corp. (https://www.siliconstudio.co.jp)
+// Copyright (c) .NET Foundation and Contributors (https://dotnetfoundation.org/ & https://stride3d.net) and Silicon Studio Corp. (https://www.siliconstudio.co.jp)
 // Distributed under the MIT license. See the LICENSE.md file in the project root for more information.
+
 #if STRIDE_GRAPHICS_API_DIRECT3D11
+
 using System;
-using SharpDX.Direct3D11;
+using Silk.NET.Core.Native;
+using Silk.NET.Direct3D11;
+
+namespace Stride.Graphics;
 
-namespace Stride.Graphics
+public unsafe partial class QueryPool
 {
-    public partial class QueryPool
+    private ComPtr<ID3D11Query>[] nativeQueries;
+
+    /// <summary>
+    ///   Gets the internal Direct3D 11 Queries.
+    /// </summary>
+    /// <remarks>
+    ///   If any of the references is going to be kept, use <see cref="ComPtr{T}.AddRef()"/> to increment the internal
+    ///   reference count, and <see cref="ComPtr{T}.Dispose()"/> when no longer needed to release the object.
+    /// </remarks>
+    internal Span<ComPtr<ID3D11Query>> NativeQueries => nativeQueries;
```

Comment:
Is the calling code allowed to modify or clear the span? If not, better use `ReadOnlySpan<T>` .

Reply:
In the original it was a `Query[]`, so it was writtable. But the only uses I can find are from `CommandList`, and it always just reads, so I'll check if it makes sense to make ir read-only